nat-lab/tests/test_teliod.py

```diff
@@ -66,7 +67,99 @@ async def test_teliod() -> None:
         assert not teliod_process.is_executing()
         assert not await is_teliod_running(connection)
 
-        # Run the hello-world command again - this time it should fail
+        # Run the get-status command again - this time it should fail
         with pytest.raises(ProcessExecError) as err:
             await connection.create_process(TELIOD_STATUS_PARAMS).execute()
         assert err.value.stderr == "Error: DaemonIsNotRunning"
+
+
+async def test_teliod_meshnet_toggle() -> None:
+    async with AsyncExitStack() as exit_stack:
+        connection = (
+            await setup_connections(exit_stack, [ConnectionTag.DOCKER_CONE_CLIENT_1])
+        )[0].connection
+
+        # Run teliod
+        teliod_process = await exit_stack.enter_async_context(
+            connection.create_process(TELIOD_START_PARAMS).run()
+        )
+
+        # Let the daemon start
+        while not await is_teliod_running(connection):
+            await asyncio.sleep(0.1)
+
+        # Run the disable-meshnet command
+        assert (
+            "executed successfully"
+            in (
+                await connection.create_process(TELIOD_DISABLE_PARAMS).execute()
+            ).get_stdout()
+        )
+
+        # Run the get-status command
+        assert (
+            '"meshnet_enabled": false'
+            in (
+                await connection.create_process(TELIOD_STATUS_PARAMS).execute()
+            ).get_stdout()
+        )
+
+        # Run the enable-meshnet command
+        assert (
+            "executed successfully"
+            in (
+                await connection.create_process(TELIOD_ENABLE_PARAMS).execute()
+            ).get_stdout()
+        )
+
+        # Run the get-status command
+        assert (
+            '"meshnet_enabled": true'
+            in (
+                await connection.create_process(TELIOD_STATUS_PARAMS).execute()
+            ).get_stdout()
+        )
+
+        assert teliod_process.is_executing()
+
+
+async def test_teliod_set_ip() -> None:
```

Comment:
Same as the previous test.

This is testing the status reporting and not really if the interface is correctly configured by the `set-ip` command.

I think these two tests can be made unit-tests (@ teliod sources), and I'd expect from integration tests it should test a more high-level-close-to-reality scenario. eg: Enable meshnet > set-ip and check that the interface is correctly configured and there is meshnet connection.

Reply:
Hmm good point, but since we are not getting any API responses in NAT-LAB, we have no IP to set, we could set the fake IP though 🤔

Reply:
> since we are not getting any API responses in NAT-LAB

We are, check my[ previous comment](https://github.com/NordSecurity/libtelio/pull/953#discussion_r1853691588).

I guess we'll need to be able to set in `teliod` the meshmap the nat-lab API provides us.